[docs] introduce Expo package API reference page (#28945)
expo/expo: docs/pages/versions/unversioned/sdk/expo.mdx

---
title: Expo

> description: Set of common functions and types for Expo and related packages.

Reviewer: ```suggestion
description: Set of common methods and types for Expo and related packages.
```

If we are going to use call functions "methods", then let's stick to that. Currently, we do use methods almost everywhere.

sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo'
packageName: 'expo'
iconUrl: '/static/images/packages/expo.png'
platforms: ['android', 'ios', 'tvos', 'web']
---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';

## Installation

<APIInstallSection hideBareInstructions />

## API

```tsx
import * as Expo from 'expo';
```

<APISection packageName="expo" />

## Common questions

### What if I want to name my main app file something other than App.js?

> You can set the `"main"` in **package.json** to any file within your project. If you do this, then you need to use `registerRootComponent`,
> `export default` will not make this component the root for the app if you are using a custom entry file.

Reviewer: ```suggestion
You can set the `"main"` in **package.json** to any file within your project. If you do this, then you need to use `registerRootComponent`. The `export default` will not make this component the root of the app if you are using a custom entry file.
```
Let's split it.

> For example, let's say you want to make **src/main.jsx** the entry file for your app - maybe you don't like having JavaScript files in the

Reviewer: ```suggestion
For example, you want to make **src/main.jsx** the entry file for your app and organize all of your app's source code in **src** directory. You can set this in **package.json**:
```

project root. First, set this in **package.json**:

```json package.json
{
  "main": "src/main.jsx"
}
```

> Then in **src/main.jsx**, make sure you call `registerRootComponent` and pass in the component you want to render at the root of the app.

Reviewer: ```suggestion
Then, in **src/main.jsx**, make sure you call `registerRootComponent` and pass in the component you want to render at the root of the app:
```

### What if I want to name my main app file something other than App.js?

You can set the `"main"` in **package.json** to any file within your project. If you do this, then you need to use `registerRootComponent`. The `export default` will not make this component the root of the app if you are using a custom entry file.

For example, you want to make **src/main.jsx** the entry file for your app and organize all of your app's source code in **src** directory. You can set this in **package.json**:

> project root. First, set this in **package.json**:

Reviewer: ```suggestion
```

I forgot to mention that we should remove this 😄
